packages/playwright-test/src/reporters/merge.ts

```diff
@@ -165,15 +199,27 @@ async function mergeEvents(dir: string, shardReportFiles: string[], printStatus:
       else if (event.method !== 'onBlobReportMetadata' && event.method !== 'onBegin')
         events.push(event);
     }
+
+    // Save information about the reports to stream their test events later.
+    reports.push({
+      idsPatcher,
+      reportFile: localPath,
+    });
   }
-  return [
-    mergeConfigureEvents(configureEvents),
-    ...projectEvents,
-    { method: 'onBegin', params: undefined },
-    ...events,
-    mergeEndEvents(endEvents),
-    { method: 'onExit', params: undefined },
-  ];
+
+  return {
+    prologue: [
+      mergeConfigureEvents(configureEvents),
+      ...projectEvents,
+      { method: 'onBegin', params: undefined },
+      ...events,
```

Comment:
What else could be here? I'd expect `events.length` to be zero.

Reply:
Good point, removed.